File: frontend/types/order.ts
export type Order = {
  id: number;
  market_id: number;
  strategy_decision_id: number | null;
  mode: "paper" | "real" | string;
  external_order_id: string | null;
  token_id: string;
  outcome: "UP" | "DOWN" | string;
  side: "BUY" | "SELL" | string;
  order_type: string;
  price: string;
  size: string;
  size_matched: string;
  status: string;
  submitted_at: string;
  updated_at: string;
  filled_at: string | null;
  raw_response: Record<string, unknown>;
  error_message: string | null;
};

File: backend/app/schemas/order.py
from datetime import datetime
from decimal import Decimal
from typing import Any

from pydantic import BaseModel, ConfigDict


class OrderResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    market_id: int
    strategy_decision_id: int | None
    mode: str
    external_order_id: str | None
    token_id: str
    outcome: str
    side: str
    order_type: str
    price: Decimal
    size: Decimal
    size_matched: Decimal
    status: str
    submitted_at: datetime
    updated_at: datetime
    filled_at: datetime | None
    raw_response: dict[str, Any]
    error_message: str | None

